src/main/java/com/commercetools/sync/products/utils/VariantReferenceReplacementUtils.java

```diff
@@ -147,10 +145,25 @@
     }
 
     private static Optional<Set<Reference<Product>>> getProductReferenceSet(@Nonnull final Attribute attribute) {
-        try {
-            return Optional.of(attribute.getValue(AttributeAccess.ofProductReferenceSet()));
-        } catch (final JsonException exception) {
-            return Optional.empty();
+        return isProductReferenceSet(attribute)
+            ? Optional.of(attribute.getValue(AttributeAccess.ofProductReferenceSet())) : Optional.empty();
+    }
+
+    static boolean isProductReference(@Nonnull final Attribute attribute) {
+        final JsonNode valueAsJsonNode = attribute.getValueAsJsonNode();
+        return !(valueAsJsonNode instanceof ArrayNode) && isValueAProductReference(valueAsJsonNode);
+    }
+
+    static boolean isProductReferenceSet(@Nonnull final Attribute attribute) {
+        final JsonNode valueAsJsonNode = attribute.getValueAsJsonNode();
+        return (valueAsJsonNode instanceof ArrayNode) && isValueAProductReference(valueAsJsonNode.elements().next());
+    }
+
+    private static boolean isValueAProductReference(@Nonnull final JsonNode valueAsJsonNode) {
+        if (valueAsJsonNode.isContainerNode()) {
+            final JsonNode typeIdNode = valueAsJsonNode.get("typeId");
```

Comment:
maybe (in another task) these kind of names should be carried out to some `CtpPlatformConstants` or something similar. Same for `"product"` references.

Reply:
In general, I don't like having separate constant "container" files which get really big by time. Especially if it's not used in different places. I might move it to such file though if I need it again.

Reply:
ok, that was just a recommendation